lang/java/reef-utils/src/main/java/org/apache/reef/util/Optional.java

```diff
@@ -35,19 +35,15 @@
   private static final long serialVersionUID = 42L;
 
   private final T value;
-  private final String valueStr;
-  private final int valueHash;
+  private static final String EMPTY_VALUE_STR = "OptionalvNothing";
+  private static final int EMPTY_VALUE_HASH = 0;
```

Comment:
Hmm, looking at the [Oracle Java8 doc](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#hashCode--), for the `hashCode()` method, it says:

> Returns the hash code value of the present value, if any, or 0 (zero) if no value is present.

Shouldn't it be implemented the same was as how it is with Java8 for us to smoothly move over to Java8 later?
Looking at it again, I've been just thinking about getting rid of the variable after all, since we can just
 
```
  public int hashCode() {
    return this.value == null ? this.value.hashCode() : 0;
  }
```
But still, please let me know if you still recommend the hashCode for an empty `Optional` to be changed.